util/progress/progressui/init.go

```diff
@@ -41,9 +42,13 @@ func init() {
 	// Make the terminal height configurable at runtime.
 	termHeightStr := os.Getenv("BUILDKIT_TTY_LOG_LINES")
 	if termHeightStr != "" {
+		fd := uintptr(os.Stdout.Fd())
 		termHeightVal, err := strconv.Atoi(termHeightStr)
 		if err == nil && termHeightVal > 0 {
-			termHeight = termHeightVal
+			winSize, err := term.GetWinsize(fd)
+			if err == nil {
+				termHeight = max(termHeight, min(termHeightVal, int(winSize.Height)-termHeight-1))
```

Comment:
I think this condition should be moved to `ttyDisplay.print()`. There you already have the terminal height for correct console (without reading `os` pkg in util what may not match the writer) and it also picks up the resizes.